Support Naming Policy

bells17 opened the pull request:
#6

ymmt2005 commented on hooks/subnamespace.go:
```diff
@@ -67,15 +70,44 @@ func (v *subNamespaceValidator) Handle(ctx context.Context, req admission.Reques
+	return admission.Allowed("")
+}
+
+func (v *subNamespaceValidator) notMatchingNamingPolicy(ctx context.Context, sn *accuratev1.SubNamespace) (bool, error) {
```
The regular expression should be pre-compiled when loaded from the configuration file,
so that no error would be returned during execution.

ymmt2005 commented on pkg/config/types.go:
```diff
@@ -10,11 +10,17 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
+type NamingPolicy struct {
```
Add a comment for godoc.

ymmt2005 commented on pkg/config/types.go:
```diff
@@ -10,11 +10,17 @@ import (
 )
 
+type NamingPolicy struct {
+	Root  string `json:"root,omitempty"`
+	Match string `json:"match,omitempty"`
```
These two are not expected to be empty, so `omitempty` should be removed.

ymmt2005 reviewed (changes requested):
Please validate the regular expressions beforehand, as commented.
Also, please add a user manual for this feature.

bells17 commented:
Thank you @ymmt2005 .
I've corrected the points you commented on.
Would you review this PR again?

> Also, please add a user manual for this feature.

I added the description and example for the naming policy setting to values.yaml.
Should we write the docs aside from the above description and example?

zoetrope reviewed (dismissed):
LGTM

ymmt2005 commented on pkg/config/types.go:
```diff
@@ -3,18 +3,32 @@ package config
 	"sigs.k8s.io/yaml"
 )
 
+// NamingPolicy represents the configuration of naming policies for Namespaces these are managed by Accurate.
```
```suggestion
// NamingPolicy represents naming policies for Namespaces created from SubNamespaces.
```

zoetrope reviewed (dismissed):
LGTM

ymmt2005 commented:
@bells17 
I'm  a little bit confused that there are almost two identical files, namely,
charts/accurate/values.yaml  and config.yaml.

Can we remove config.yaml?

bells17 commented:
@ymmt2005 Thank you for your reviewing and I removed config.yaml.
Would you check it?

ymmt2005 reviewed (approved):
LGTM
